src/haxelib/api/Installer.hx

```diff
@@ -795,8 +795,18 @@ class Installer {
 
 			final currentBranch = vcsBranchesByLibraryName[library];
 
-			// TODO check different urls as well
-			if (branch != null && (!wasUpdated || currentBranch != branch)) {
+			var libUrl = vcs.getReproducibleVersion(libPath).url;
+
+			if (url != null && libUrl != null && url != libUrl)
+			{
+				if (!userInterface.confirm('Update remote url: "${libUrl}" with "${url}"')) {
```

Comment:
This introduces a new cli interaction that wasn't present before, it may be best to simply overwrite unless there are uncommitted changes.